cobalt/user_agent_string_test.cc

```diff
@@ -0,0 +1,439 @@
+// Copyright 2024 The Cobalt Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "cobalt/user_agent_string.h"
+
+#include <map>
+
+#include "cobalt/user_agent_platform_info.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+namespace cobalt {
+
+namespace {
+
+UserAgentPlatformInfo CreateEmptyPlatformInfo() {
+  UserAgentPlatformInfo platform_info;
+  platform_info.set_starboard_version("");
+  platform_info.set_os_name_and_version("");
+  platform_info.set_original_design_manufacturer("");
+  platform_info.set_device_type("UNKNOWN");
+  platform_info.set_chipset_model_number("");
+  platform_info.set_model_year("");
+  platform_info.set_firmware_version("");
+  platform_info.set_brand("");
+  platform_info.set_model("");
+  platform_info.set_aux_field("");
+  platform_info.set_javascript_engine_version("");
+  platform_info.set_rasterizer_type("");
+  platform_info.set_evergreen_version("");
+  platform_info.set_evergreen_type("");
+  platform_info.set_evergreen_file_type("");
+  platform_info.set_cobalt_version("");
+  platform_info.set_cobalt_build_version_number("");
+  platform_info.set_build_configuration("");
```

Comment:
But many of these are not needed because std::strings are constructed empty by default.

What you can do is to leave an `ASSERT_STREQ()` or similar to make sure that certain member variables are indeed empty (documenting via assertions).

Reply:
Nope, when `UserAgentPlatformInfo platform_info;` is initialized, the class constructor invokes `void InitializeUserAgentPlatformInfoFields(UserAgentPlatformInfo& info)` to initialize all the UA properties, and the std::string properties might not be constructed empty by default.

Reply:
Hmmm I see, what a turn of events. Could you leave a comment here as to why it's needed then? Something like '// UserAgentPlatformInfo is not constructed empty by default.` in l.27? Thanks!